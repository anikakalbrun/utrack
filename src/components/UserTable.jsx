import React from 'react';
import {array} from 'prop-types';
import {Table} from 'antd';
import 'antd/dist/antd.css';

UserTable.propTypes = {
    data: array
};

UserTable.defaultProps = {};

const columns = [{
    title: 'User Id',
    dataIndex: 'id',
    key: 'id',
}, {
    title: 'Birthday',
    dataIndex: 'birthday',
    key: 'birthday',
}, {
    title: 'Spend',
    dataIndex: 'spend',
    key: 'spend',
}, {
    title: 'Region',
    dataIndex: 'region',
    key: 'region',
}, {
    title: 'Gender',
    dataIndex: 'gender',
    key: 'gender',
}
];

export default function UserTable({data}) {
    return (
        <div className='width100 p-50 tableContainer'>
            <Table dataSource={data} columns={columns} rowKey="id"/>
        </div>
    );
}