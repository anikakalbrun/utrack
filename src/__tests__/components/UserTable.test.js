import React from 'react';
import UserTable from '../../components/UserTable';

let wrapper;

describe('UserTable', () => {

    beforeEach(() => {
        wrapper = shallow(<UserTable data={[]}/>);
    });

    it('renders correctly a wrapper div', () => {
        expect(wrapper.find('div.width100.p-50.tableContainer').length).toEqual(1);
        expect(wrapper.find('Table').length).toEqual(1);
        expect(wrapper.find('Table').props().columns).toMatchSnapshot();
    });

    it('renders the Table component', () => {
        expect(wrapper.find('Table').length).toEqual(1);
    });

    it('passes to the Table a columns property', () => {
        expect(wrapper.find('Table').props().columns).toMatchSnapshot();
    });

    it('passes to the Table a rowKey="id" property', () => {
        expect(wrapper.find('Table').props().rowKey).toEqual('id');
    });

    it('passes to the Table a dataSource property', () => {
        expect(wrapper.find('Table').props().dataSource).toEqual([]);
    });
});
