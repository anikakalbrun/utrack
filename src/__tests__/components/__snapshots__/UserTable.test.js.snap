// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable passes to the Table a columns property 1`] = `
Array [
  Object {
    "dataIndex": "id",
    "key": "id",
    "title": "User Id",
  },
  Object {
    "dataIndex": "birthday",
    "key": "birthday",
    "title": "Birthday",
  },
  Object {
    "dataIndex": "spend",
    "key": "spend",
    "title": "Spend",
  },
  Object {
    "dataIndex": "region",
    "key": "region",
    "title": "Region",
  },
  Object {
    "dataIndex": "gender",
    "key": "gender",
    "title": "Gender",
  },
]
`;

exports[`UserTable renders correctly a wrapper div 1`] = `
Array [
  Object {
    "dataIndex": "id",
    "key": "id",
    "title": "User Id",
  },
  Object {
    "dataIndex": "birthday",
    "key": "birthday",
    "title": "Birthday",
  },
  Object {
    "dataIndex": "spend",
    "key": "spend",
    "title": "Spend",
  },
  Object {
    "dataIndex": "region",
    "key": "region",
    "title": "Region",
  },
  Object {
    "dataIndex": "gender",
    "key": "gender",
    "title": "Gender",
  },
]
`;
